### PR TITLE
azure_rm_virtualmachine: parallelise tests

### PR DESCRIPTION
**Pull request by spmp**
##### SUMMARY
Parallelise tests for `azure_rm_virtualmachine` to reduce timeouts, and enable quicker testing of new components.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
azure_rm_virtualmachine

##### ANSIBLE VERSION
```
2.7
```

##### ADDITIONAL INFORMATION
`accept_terms` tests are disabled in the `inventory` file.
The `azure_rm_virtualmachine_facts` tests removed.

### CODE REVIEW

**Comment by spmp** [comment]
We are green in record time! @Fred-sun this is ready for review by @jborean93, @nitzmahone, @mattclay, @zikalino, and others

**Comment by yuwzho** [comment]
@mattclay and @nitzmahone to review the test

File: test/integration/targets/azure_rm_virtualmachine/runme.sh
Comment (mattclay) [inline test/integration/targets/azure_rm_virtualmachine/runme.sh]: No need to add `-vvv` here, since `ansible-test` will pass along `-v` usage to `runme.sh`. As long as the script uses `"$@"` as below it should work properly.

File: test/integration/targets/azure_rm_virtualmachine/basic.yaml
Comment (mattclay) [inline test/integration/targets/azure_rm_virtualmachine/basic.yaml]: I realize these tests were already disabled in the original tests, but they could be made to work by setting the `ANSIBLE_TEST_REMOTE_INTERPRETER` environment variable to an empty string, or to the path to the interpreter:

```yaml
    environment:
        ANSIBLE_TEST_REMOTE_INTERPRETER: ""
```

This will disable the coverage injector for the delegated hosts.

File: test/integration/targets/azure_rm_virtualmachine/accept_terms.yaml
Comment (mattclay) [inline test/integration/targets/azure_rm_virtualmachine/accept_terms.yaml:50]: Add a trailing newline here and to other files that lack one.

**Review by mattclay — CHANGES_REQUESTED** [review]

File: test/integration/targets/azure_rm_virtualmachine/basic.yaml
Comment (mattclay) [inline test/integration/targets/azure_rm_virtualmachine/basic.yaml]: This is causing integration test failures, as add_host cannot be used with the free strategy:

https://app.shippable.com/github/ansible/ansible/runs/81844/73/console

Which means we won't be able to run this part of the test in parallel.

**Comment by Fred-sun** [comment]
@spmp Thanks for the  contribution, Could you help to resolve the confliction file? Thanks!

File: test/integration/targets/azure_rm_virtualmachine/accept_terms.yaml
Comment (yungezz) [inline test/integration/targets/azure_rm_virtualmachine/accept_terms.yaml:50]: if you don't care about deletion result, pls add async the deletion

File: test/integration/targets/azure_rm_virtualmachine/os_disk_size.yaml
Comment (yungezz) [inline test/integration/targets/azure_rm_virtualmachine/os_disk_size.yaml:3]: It's too expensive to create a vm for each of test scenario, since vm creation/deletion is time  consuming, and there's  core limitation in test subscription. Suggest to combine some of those test, eg. image  disk  size with no_public ip test together.

**Review by yungezz — CHANGES_REQUESTED** [review]
reduce vm creation/deletion

**Comment by yungezz** [comment]
@zikalino

File: test/integration/targets/azure_rm_virtualmachine/basic.yaml
Comment (spmp) [reply test/integration/targets/azure_rm_virtualmachine/basic.yaml]: I am sorry I do not know enough to get this right. Can you complete this please. I added debug to look at the output as I had trouble resolving the IP address in the past with this test.

**Comment by spmp** [comment]
I will do my best to resolve all issues tomorrow (if I get time) then I have to hand over responsibility to someone else. As I have to wrap this work up ASAP 8(
 I have local Azure access again, so should be able to test most of this before pushing to your CI and creating a heap of noise.

File: test/integration/targets/azure_rm_virtualmachine/os_disk_size.yaml
Comment (spmp) [reply test/integration/targets/azure_rm_virtualmachine/os_disk_size.yaml:3]: The motivation here is to increase the number of possible tests without timeout. AS `devel` stands currently we are at the cusp of timeouts with the number of tests.
 This module needs many more tests and what is present in this PR is still a somewhat minimal set.
Where possible VM's have been preserved, yet testing with managed disk, with specified OS disk size, and market place images require new VM's.
Essentially each _line_ of VM's is separated in the `inventory`.

File: test/integration/targets/azure_rm_virtualmachine/accept_terms.yaml
Comment (spmp) [reply test/integration/targets/azure_rm_virtualmachine/accept_terms.yaml:50]: Done

Comment (spmp) [reply test/integration/targets/azure_rm_virtualmachine/accept_terms.yaml:50]: Done

**Comment by mattclay** [comment]
@spmp I recommend waiting until tomorrow to try running CI for Azure modules. Azure is still experiencing an outage that appears to be impacting CI:

https://azure.microsoft.com/en-us/status/

**Comment by spmp** [comment]
@good-people! I am out of time for Ansible, please let me know if you need write access (and don't have it) too my repo to make changes to see this over the line.
 I believe that re-enabling the commented tests is out of scope for this PR, but could be achieved with a custom shell command (`ssh cat <that file>`) if the hosts were setup using ssh keys. Feels like the wrong way to me.

Good luck 8)

**Comment by Fred-sun** [comment]
@zikalino this PR exist conflicting files. Could you help take a look this PR? Thanks!

**Comment by Fred-sun** [comment]
Which one will keep working on this PR?

**Comment by spmp** [comment]
@Fred-sun I cannot sorry.

**Comment by Fred-sun** [comment]
@yungezz

**Comment by Fred-sun** [comment]
@zikalino Could you help take a look this PR? Thanks!

**Comment by Fred-sun** [comment]
Which one will continue contribute to this issue?

**Comment by Fred-sun** [comment]
need new contributor , @zikalino Could you help this? Thanks!

**Comment by spmp** [comment]
Sorry I cannot put hours into this ATM.

**Comment by Fred-sun** [comment]
@zikalino @yuwzho @yungezz

**Comment by Fred-sun** [comment]
kindly ping

**Comment by Fred-sun** [comment]
@zikalino @yuwzho @yungezz Please help take a look this PR! There is no contributor continue on this！ Thanks!

**Comment by Fred-sun** [comment]
@yungezz @yuwzho

**Comment by Fred-sun** [comment]
kindly ping

**Comment by Fred-sun** [comment]
@yuwzho  This PR has not been updated for more than three months,  and exist conflicting files.  Can it be closed temporarily? when get the new demand to reopen again! Thanks!

**Comment by Fred-sun** [comment]
kindly ping

**Comment by Fred-sun** [comment]
@yuwzho  Could we can closed this PR?  The contributor disappeared and existed conflicting files! Thank you very much!

**Comment by Fred-sun** [comment]
@yuwzho advice closed this, Re-open or submit new PR when needed! Thanks a lot!

**Comment by yuwzho** [comment]
close_me

Close this one temporally as we are addressing all VM issues, will refine the test as well

**Pull request closed without merging**
